### PR TITLE
[router] Alpini dead code removal

### PR DESCRIPTION
**Pull request by FelixGV**
Besides dead code removal, also moved a few functions from Aplini's CollectionUtil to Venice's CollectionUtils.

## How was this PR tested?

CI.

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: internal/alpini/netty4/alpini-netty4-base/src/main/java/com/linkedin/alpini/netty4/misc/NettyUtils.java [hunk 1]
```diff
@@ -32,143 +16,8 @@ private NettyUtils() {
...
-    return mode().socketChannel();
-  }
-
-  public static Class<? extends ServerSocketChannel> serverSocketChannel() {
-    return mode().serverSocketChannel();
-  }
-
-  public static Class<? extends DatagramChannel> datagramChannel() {
-    return mode().datagramChannel();
+    return in.readBytes(length);
```
Comment (nisargthakkar) [inline internal/alpini/netty4/alpini-netty4-base/src/main/java/com/linkedin/alpini/netty4/misc/NettyUtils.java:20]: 😂

File: internal/venice-client-common/src/main/java/com/linkedin/venice/utils/CollectionUtils.java [hunk 2]
```diff
@@ -112,4 +120,92 @@ public static Map<String, String> convertCharSequenceMapToStringMap(Map<CharSequ
   public static <K, V> Map<K, V> substituteEmptyMap(Map<K, V> map) {
     return map == null || map.isEmpty() ? Collections.emptyMap() : map;
   }
+
+  public static <K, V> MapBuilder<K, V> mapBuilder() {
```
Comment (nisargthakkar) [inline internal/venice-client-common/src/main/java/com/linkedin/venice/utils/CollectionUtils.java:124]: This is a great addition. Thanks

**Review by nisargthakkar — DISMISSED** [review]
LGTM

**Comment by nisargthakkar** [comment]
I'm a bit confused why the Alpini checks didn't run. Maybe we should make them run on every PR instead of the path based filtering?

**Comment by nisargthakkar** [comment]
They did run and now they seem to be quite quick. We should definitely enable them for every build

**Comment by nisargthakkar** [comment]
The functional tests are looking sketchy. Looks like no actual test ran. We should take a look at what's up with that

**Comment by FelixGV** [comment]
I was thinking of enabling them on every run, yeah... will have to check what happened to the functional ones.

**Review by nisargthakkar — APPROVED** [review]

**Pull request merged**